# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: toolkit/components/remotebrowserutils/tests/browser/coop_header.sjs
# Commit: 5686e9cc48b2
# Full Hash: 5686e9cc48b26a8ec550f94091d5b8b4ed840f7d
# Author: Valentin Gosu <valentin.gosu@gmail.com>
# Date: 2019-10-22 09:46:06
# Regressor Bug: 1579992
# File Overlap Count: 1
# Description:
#   Bug 1579992 - Load pages into new webCOOP+COEP process type r=nika
#   
#   * This patch makes pages with the `OPENER_POLICY_SAME_ORIGIN_EMBEDDER_POLICY_REQUIRE_CORP` policy load into a special `webCOOP+COEP={pageOrigin}` remote type.
#   * Adds `E10SUtils.WEB_REMOTE_COOP_COEP_TYPE_PREFIX="webCOOP+COEP="`
#   * When a COOP process switch occurs and the target page doesn't have this policy, we pass a `preferredRemoteType="web"` into `E10SUtils.getRemoteTypeForPrincipal` ensuring that we correctly get a different `remoteType`
# ==============================================================================

diff -r fd0d2f340380 -r 5686e9cc48b2 toolkit/components/remotebrowserutils/tests/browser/coop_header.sjs
--- a/toolkit/components/remotebrowserutils/tests/browser/coop_header.sjs	Mon Oct 21 16:34:43 2019 +0000
+++ b/toolkit/components/remotebrowserutils/tests/browser/coop_header.sjs	Mon Oct 21 16:56:00 2019 +0000
@@ -1,22 +1,24 @@
 function handleRequest(request, response)
 {
-  response.setStatusLine(request.httpVersion, 200, "OK");
+  Components.utils.importGlobalProperties(["URLSearchParams"]);
+  let query = new URLSearchParams(request.queryString);
 
-  let qs = request.queryString.replace(/\./g, '');
+  response.setStatusLine(request.httpVersion, 200, "OK");
 
   let isDownloadPage = false;
   let isDownloadFile = false;
-  if (qs.length > 0) {
-    qs.split("&").forEach(param => {
-      if (param === "downloadPage") {
-        isDownloadPage = true;
-      } else if (param === "downloadFile") {
-        isDownloadFile = true;
-      } else if (param.length > 0) {
-        response.setHeader("Cross-Origin-Opener-Policy", unescape(param), false);
-      }
-    });
-  }
+
+  query.forEach((value, name) => {
+    if (name === "downloadPage") {
+      isDownloadPage = true;
+    } else if (name === "downloadFile") {
+      isDownloadFile = true;
+    } else if (name ==  "coop") {
+      response.setHeader("Cross-Origin-Opener-Policy", unescape(value), false);
+    } else if (name == "coep") {
+      response.setHeader("Cross-Origin-Embedder-Policy", unescape(value), false);
+    }
+  });
 
   let downloadHTML = "";
   if (isDownloadPage) {